core/src/impl/Kokkos_Error.hpp

```diff
@@ -164,9 +164,30 @@ class RawMemoryAllocationFailure : public std::bad_alloc {
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
 
+#if defined(KOKKOS_ENABLE_CUDA) && defined(__CUDA_ARCH__)
+
+#if defined(__APPLE__)
+// cuda_abort does not abort when building for macOS.
+#define KOKKOS_IMPL_ABORT_NORETURN
+#else
+// cuda_abort aborts when building for other platforms than macOS
+#define KOKKOS_IMPL_ABORT_NORETURN [[noreturn]]
+#endif
+
+#elif defined(KOKKOS_ENABLE_HIP) && defined(__HIP_DEVICE_COMPILE__)
+// HIP does not abort
```

Comment:
This changed recently, see https://github.com/kokkos/kokkos/blob/89e9d26d58f94547ba219a71ed5bde4d935715cd/core/src/HIP/Kokkos_HIP_Abort.hpp#L56-L63

Reply:
oh man. I was about to merge it :-(

Reply:
I would be fine with addressing that in a follow-up pull request.

Reply:
I am currently seeing
```
/tmp/kokkos/core/src/HIP/Kokkos_HIP_Abort.hpp:64:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
```
if trying to mark the function with `[[noreturn]]` so we might just postpone fixing it.